src/ucp/rndv/rndv.c

```diff
@@ -1498,6 +1516,7 @@ UCS_PROFILE_FUNC_VOID(ucp_rndv_put_pipeline_frag_get_completion, (self),
     ucp_request_t *fsreq = freq->send.rndv_get.rreq;
 
     /* get completed on memtype endpoint to stage on host. send put request to receiver*/
+    ucp_rkey_destroy(freq->send.rndv_get.rkey);
```

Comment:
freq->send.rndv_get.rkey is NULL cuda case right? may because you packed key for cuda also at line:798, it didn't crash in the gtests

Reply:
Added a NULL check for freq->send.rndv_get.rkey